src/web/server/app/services/apiGroups.ts

```diff
@@ -0,0 +1,156 @@
+  INSERT_GROUPID_USERID,
+  SELECT_GROUPUSERS_GROUPID,
+} from '../utils/queries';
+
+import { client } from '../utils/client';
+import * as apiServices from './api';
+
+/**
+ * Assign user to group
+ * @param userId - The userId of the user
+ * @param groupId - The groupId of the group
+ */
+export const addUserGroup = async (userId: string, groupId: string) => {
+  const params = [userId, groupId];
+  await client.execute(INSERT_USERID_GROUPID, params, { prepare: true });
+  await client.execute(INSERT_GROUPID_USERID, params, { prepare: true });
+};
+
+/**
+ * Verifies if userId can view cameraId based on group
+ * @param cameraId - The cameraId
+ * @param userId - The userId that wishes to view camera
+ * @returns boolean - True is user can view camera
+ */
+export const verifyUserCameraGroup = async (
+  userId: string,
+  cameraId: string
+) => {
+  const camerasResult = await retrieveUserGroupCameras(userId);
+  if (camerasResult === undefined || camerasResult.length === 0) {
+    return undefined;
+  }
+  return camerasResult.some(cameraIdElement => {
+    return cameraIdElement === cameraId;
+  });
+};
+
+/**
+ * Retrieve all group ids
+ * @returns ResultSet - Contains rows of groupIds
+ */
+export const retrieveGroups = async () => {
+  return client.execute(SELECT_GROUPUSERS_GROUPID, [], {
+    prepare: true,
+  });
+};
+
+/**
+ * Retrieve groupIds from database that user belongs to
+ * @param userId - The user id of the user
+ * @returns ResultSet - Contains rows of groupIds belonging to userId
+ */
+export const retrieveUserGroups = async (userId: string) => {
+  return client.execute(SELECT_GROUPID_USERID, [userId], {
+    prepare: true,
+  });
+};
+
+/**
+ * Retrieve userIds from database that belong in groupId
+ * @param groupId - The group id of the group
+ * @returns ResultSet - Contains rows of userIds belonging to groupId
+ */
+export const retrieveGroupUsers = async (groupId: string) => {
+  return client.execute(SELECT_USERID_GROUPID, [groupId], {
+    prepare: true,
+  });
+};
+
+/**
+ * Retrieve cameraIds from users in database that belong in groupId
+ * @param groupId - The group id of the group
+ * @returns Array<string> - Contains cameraIds of users belonging to groupId
+ */
+export const retrieveGroupCameras = async (groupId: string) => {
+  const groupUsers = await retrieveGroupUsers(groupId);
+  if (groupUsers === undefined || groupUsers.rows.length === 0) {
+    return undefined;
+  }
+  const groupCameras = groupUsers.rows.reduce(
+    async (oldCollection: Promise<string[]>, row: any) => {
+      const collection = await oldCollection;
+      const userCameras = await apiServices.retrieveCameraIds(row['user_id']);
+      if (userCameras !== undefined && userCameras.rows.length > 0) {
+        collection.push(
+          ...userCameras.rows.map((row: any) => {
+            return row['camera_id'];
+          })
+        );
+      }
+      return collection;
+    },
+    new Promise((resolve, reject) => resolve([]))
```

Comment:
What is this for?

Reply:
This is the initial collection for reduce. A promise for an empty array

Reply:
ok